frontend/BoardGUI.py

```diff
@@ -1,10 +1,12 @@
 import pygame as pg
-
+import time
+import json
 from backend.Board import Board
 from backend.Cell import Color
 import sys
-vec2 = pg.math.Vector2 
 
+vec2 = pg.math.Vector2 
+history_path = '/Users/suryanshpatel/Projects/9mens morris Software enginnering/Nine-Mens-Morris/resources/history/game_history.json'
```

Comment:
If we dynamically save files, this could either grab the most recent file saved (easy approach), allow the user to pick their own file (would likely need another library, unless pygame has dialogue boxes that I've missed built-in), or an extra screen where we print the user's options and let them select the file to open.

Reply:
@suryansh-max @raspberrymilkyway I have made the changes for the dynamic path here while working on the unit test and acceptance tests. You guys can check on my new commit in this PR.